src/hermes/commands/harvest/util/remote_harvesting.py

```diff
@@ -0,0 +1,93 @@
+# SPDX-FileCopyrightText: 2025 OFFIS e.V.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# SPDX-FileContributor: Stephan Ferenz
+# SPDX-FileContributor: Aida Jafarbigloo
+
+import pathlib
+import re
+import requests
+import tempfile
+import typing as t
+import os
+
+def normalize_url(path: str) -> str:
+    """Normalize a given URL by correcting backslashes and fixing malformed HTTPS."""
+    corrected_url = path.replace("\\", "/")
+    return corrected_url.replace("https:/", "https://")
+
+
+def fetch_metadata_from_repo(repo_url: str, filename: str) -> t.Optional[pathlib.Path]:
```

Comment:
This method makes multiple HTTP requests. I think it would be nice to use the hermes user agent, just to let the services know who we are. You can do something like:

```python
from hermes.utils import hermes_user_agent

session = requests.Session()
session.headers.update({"User-Agent": hermes_user_agent})
```

then use the session to make the requests:

```python
session.get(api_url)
```